feat(gateway): Datakit spans and tracing in Debugger (#5590)
Kong/developer.konghq.com: app/observability/datakit-debugger.md

  - attribute: "`proxy.kong.datakit.node.status`"
    description: The terminal status of the node after execution.
    values: "`complete`, `fail`, `cancel`"
{% endtable %}
<!--vale on-->

## Tracing events

Tracing events capture the full lifecycle of each node, including input and output values.
Each event corresponds to a specific point in a node's execution and includes the data the node was working with at that moment.

{:.info}
> Large values may be omitted from tracing events if they exceed the capture size limit.

Each tracing event includes the following fields:

<!--vale off-->
{% table %}
columns:
  - title: Field
    key: field
  - title: Description
    key: description
rows:
  - field: "`name`"
    description: The name of the Datakit node.
  - field: "`type`"
    description: The type of the Datakit node (for example, `jq`, `call`, `jwt_decode`).
  - field: "`action`"
    description: The lifecycle action that generated this event. See [Actions](#actions).
  - field: "`timestamp`"
    description: The time the event occurred.
  - field: "`value`"

>     description: The input or output value associated with this action. Present for `run`, `complete`, and `fail` actions. May be absent for large values that exceed capture limits.

Author: This seems wrong, will verify.

---
title: Using Debugger for Datakit
description: 'Use {{site.konnect_short_name}} Debugger to inspect Datakit node execution: trace node spans, view I/O values per node lifecycle event, and understand how sanitization rules apply to Datakit tracing data.'
breadcrumbs:
  - /observability/
  - /observability/debugger/
content_type: reference
layout: reference
search_aliases:
  - active tracing
  - debugger
products:
    - observability
    - konnect
    - gateway
works_on:
    - konnect
tags:
  - tracing
  - debugger

related_resources:
  - text: "{{site.konnect_short_name}} Debugger"
    url: /observability/debugger/
  - text: Datakit plugin
    url: /plugins/datakit/
  - text: Datakit in {{site.base_gateway}}
    url: /gateway/datakit/

min_version:
  gateway: '3.15'
---

When you run a [Debugger session](/observability/debugger/) on a {{site.base_gateway}} instance that uses the [Datakit plugin](/plugins/datakit/), Debugger captures tracing data for each Datakit node in the pipeline.

> This gives you visibility into how data flows through your Datakit configuration.
> You can see which nodes ran, what values they received and produced, and where errors or skips occurred.
> 

Reviewer: ```suggestion
When you run a [Debugger session](/observability/debugger/) on a {{site.base_gateway}} instance that uses the [Datakit plugin](/plugins/datakit/), Debugger captures tracing data for each Datakit node in the pipeline.
This gives you visibility into how data flows through your Datakit configuration, which nodes ran, what values they received and produced, and where errors or skips occurred.

```

Datakit tracing provides debugging detail without exposing sensitive data.
{{site.base_gateway}} tracks values as they move through the Datakit workflow and applies redaction and sanitization rules before sending trace data to {{site.konnect_short_name}}.

## Enable tracing event collection

> To collect Datakit tracing events, start a {{site.konnect_short_name}} Debugger session with body payload capture.

Reviewer: ```suggestion
To collect Datakit tracing events, start a {{site.konnect_short_name}} Debugger session with body payload capture enabled.
```

If header-only payload capture is enabled, node spans will be recorded but tracing events won't be uploaded.

## Datakit node spans

For each Datakit node that executes, Debugger creates a span named `kong.datakit.node.NODE_NAME`, where `NODE_NAME` is the name you assigned to the node in your Datakit configuration.

Node spans capture metadata about node execution but don't contain node input or output values.
Values are only captured in the separate Datakit tracing event payload.

> If a node is skipped due to branch routing rules, it normally won't create a span because execution hasn't started.

Reviewer: ```suggestion
If a node is skipped due to branch routing rules, it won't create a span because execution hasn't started.
```
I feel like normally here feels like it might happen
